Core/include/Acts/Visualization/EventDataVisualization.hpp

```diff
@@ -0,0 +1,209 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2020 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Geometry/Polyhedron.hpp"
+#include "Acts/Surfaces/CylinderBounds.hpp"
+#include "Acts/Surfaces/CylinderSurface.hpp"
+#include "Acts/Surfaces/Surface.hpp"
+#include "Acts/Surfaces/detail/FacesHelper.hpp"
+#include "Acts/Utilities/Definitions.hpp"
+#include "Acts/Utilities/Helpers.hpp"
+#include "Acts/Utilities/UnitVectors.hpp"
+#include "Acts/Visualization/GeometryVisualization.hpp"
+#include "Acts/Visualization/IVisualization.hpp"
+
+#include <optional>
+
+namespace Acts {
+namespace Visualization {
+
+/// Helper to find the egen values and corr angle
+///
+/// @param covariance The covariance matrix
+static inline std::array<double, 3> decomposeCovariance(
```

Comment:
The functions in this file actually have numeric results, so that's something that could get it's own unit test and actually compare numeric results with reference values.

Reply:
Is specially tested now.